src/map/pc.c

```diff
@@ -10163,21 +10167,22 @@ int pc_checkitem(struct map_session_data *sd)
 				sd->status.cart[i].unique_id = itemdb->unique_id(sd);
 		}
 
-		for( i = 0; i < MAX_STORAGE; i++ ) {
-			id = sd->status.storage.items[i].nameid;
+		for (i = 0; i < VECTOR_LENGTH(sd->storage); i++) {
+			struct item *it = &VECTOR_INDEX(sd->storage, i);
+			id = it->nameid;
 
 			if (!id)
 				continue;
 
 			if( id && !itemdb_available(id) ) {
-				ShowWarning("Removed invalid/disabled item id %d from storage (amount=%d, char_id=%d).\n", id, sd->status.storage.items[i].amount, sd->status.char_id);
-				storage->delitem(sd, i, sd->status.storage.items[i].amount);
+				ShowWarning("Removed invalid/disabled item id %d from storage (amount=%d, char_id=%d).\n", id, it->amount, sd->status.char_id);
+				storage->delitem(sd, i, it->amount);
```

Comment:
See the note in atcommand clearstorage on using `storage->delitem()` inside a loop on `VECTOR_LENGTH(sd->storage)`. The same issue occurs here